src/firmware/app/control/trajectory_planner.h

```diff
@@ -1,303 +1,196 @@
 #pragma once
 
+#include <stdbool.h>
+
+#include "firmware/app/control/trajectory_planner_impl.h"
```

Comment:
We should not include the `impl` here or it will  be transitively included in anything that includes this header file (thus defeating the whole purpose). We should only have it in the `.c` file.

Reply:
The TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS is defined in the impl.h

If the .h is removed then we don't have access to that define. If we move the define to trajectory_planner.h then we need to include it in impl.h which causes a circular dependency

Reply:
As per in-person convo, I've removed the impl.h file from the tp header